.github/workflows/updoc.yml

```diff
@@ -0,0 +1,61 @@
+name: Updoc
+
+on:
+  workflow_call:
+    secrets:
+      UPBOUND_CI_PROD_BUCKET_SA:
+        required: true
+
+env:
+  GO_VERSION: '1.19'
+
+jobs:
+  upload:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+          ref: '${{ inputs.version }}'
+
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Find the Go Build Cache
+        id: go
+        run: echo "cache=$(make go.cachedir)" >> $GITHUB_OUTPUT
+
+      - name: Cache the Go Build Cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.go.outputs.cache }}
+          key: ${{ runner.os }}-build-updoc-artifacts-${{ hashFiles('**/go.sum') }}
+          restore-keys: ${{ runner.os }}-build-updoc-artifacts-
+
+      - name: Cache Go Dependencies
+        uses: actions/cache@v3
+        with:
+          path: .work/pkg
+          key: ${{ runner.os }}-updoc-pkg-${{ hashFiles('**/go.sum') }}
+          restore-keys: ${{ runner.os }}-updoc-pkg-
+
+      - name: Create Service Account File
+        env:
+          GAC: ${{ secrets.UPBOUND_CI_PROD_BUCKET_SA }}
+        run: |
+          echo "${GAC}" > sa.json
+
+      - name: Upload Docs
+        env:
+          GOOGLE_APPLICATION_CREDENTIALS: sa.json
+        run: |
+          if [[ "${GITHUB_REF##*/}" == release-* ]]; then
+            echo "Publishing Docs for ${GITHUB_REPOSITORY#*/}, v${GITHUB_REF#"refs/heads/release-"}"
```

Comment:
Let's declare two new environment variables (e.g., `PROVIDER_NAME` & `VER_MAJOR_MINOR`) for `${GITHUB_REPOSITORY#*/}` and `v${GITHUB_REF#"refs/heads/release-"}` so that we will be using these two variables for both `echo` and `updoc`.